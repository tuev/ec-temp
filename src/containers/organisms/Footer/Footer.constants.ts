import { IFooterCategory } from './Footer.types'
export const FooterItem: IFooterCategory[] = [
  {
    category: 'question',
    title: 'Questions',
    items: [
      {
        key: 'help',
        href: '/help',
        title: 'Help',
      },
      {
        key: 'track',
        href: '/trackorder',
        title: 'Track Order',
      },
      {
        key: 'return',
        href: '/returns',
        title: 'Return',
      },
    ],
  },
  {
    category: 'category',
    title: "What's in store",
    items: [
      {
        key: 'women',
        href: '/productlist?category=women',
        title: 'Women',
      },
      {
        key: 'men',
        href: '/productlist?category=men',
        title: 'Men',
      },
      {
        key: 'all',
        href: '/productlist',
        title: 'Product A to Z',
      },
    ],
  },
]

export default FooterItem
