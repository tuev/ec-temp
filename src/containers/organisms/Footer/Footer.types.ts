export interface IFooterItem {
  key: string
  title: string
  href: string
}

export interface IFooterCategory {
  category: string
  items: IFooterItem[]
  title: string
}

export type FooterProp = {
  items?: IFooterCategory[]
}
