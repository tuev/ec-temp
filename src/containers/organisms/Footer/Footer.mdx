# Stories mdx for Footer

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Footer.stories.tsx'
import Footer from './Footer.tsx'

<Story name="basic" />