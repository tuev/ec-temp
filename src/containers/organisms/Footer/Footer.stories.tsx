/**
 *
 *  Stories for Footer
 *
 */

import React from 'react'
import { action } from '@storybook/addon-actions'
import Footer from './Footer'

export default {
  component: Footer,
  includeStories: ['basic'],
  title: 'Footer',
}

export const basic = () => (
  <div>
    <Footer />
  </div>
)
