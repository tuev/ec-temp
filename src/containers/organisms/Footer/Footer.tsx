/**
 *
 * Footer
 *
 */

import React, { FC } from 'react'
import { Grid, Link } from '@material-ui/core'
import Wrapper from 'components/atoms/Wrapper'
import Typography from 'components/atoms/Typography'
import footerItems from './Footer.constants'
import { IFooterItem, IFooterCategory, FooterProp } from './Footer.types'

const Footer: FC<FooterProp> = (props: FooterProp) => {
  const { items = footerItems } = props

  return (
    <Wrapper
      p="1rem"
      width={1}
      display="flex"
      justifyContent="center"
      alignItems="center"
    >
      <Grid container={true}>
        {items.map((item: IFooterCategory) => (
          <Grid item={true} xs={12} sm={6} md={3} key={item.category}>
            <Wrapper p="1rem">
              <Typography customvariant="header3">{item.title}</Typography>
              <Wrapper mt="1rem" display="flex" flexDirection="column">
                {item.items.map((child: IFooterItem) => (
                  <Link href={child.href} key={child.key}>
                    <Typography customvariant="body1" variant="overline">
                      {child.title}
                    </Typography>
                  </Link>
                ))}
              </Wrapper>
            </Wrapper>
          </Grid>
        ))}
        <Grid item={true} xs={12} sm={6} md={3}>
          <Wrapper p="1rem">
            <Typography customvariant="header3">Contact us</Typography>
            <Wrapper mt="1rem" display="flex" flexDirection="column">
              <Wrapper>
                <Typography
                  customvariant="body1"
                  animation={1}
                  variant="overline"
                >
                  Facebook
                </Typography>
              </Wrapper>
              <Wrapper>
                <Typography
                  customvariant="body1"
                  animation={1}
                  variant="overline"
                >
                  Twitter
                </Typography>
              </Wrapper>
            </Wrapper>
          </Wrapper>
        </Grid>
        <Grid item={true} xs={12} sm={6} md={3}>
          <Wrapper p="1rem">
            <Typography>&copy;2020 shopify</Typography>
          </Wrapper>
        </Grid>
      </Grid>
    </Wrapper>
  )
}

export default Footer
