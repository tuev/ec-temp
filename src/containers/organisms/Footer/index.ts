/**
 *
 * Index for Footer
 *
 */

import Footer from './Footer'

export default Footer
