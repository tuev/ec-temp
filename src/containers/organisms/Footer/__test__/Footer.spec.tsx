/**
 *
 * Tests for Footer
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import Footer from '..'

describe('App Footer', () => {
  it('Test get default Footer no params ', () => {
    // test snapshot
    const wrapper = render(<Footer />)
    expect(wrapper).toMatchSnapshot()
  })
})
