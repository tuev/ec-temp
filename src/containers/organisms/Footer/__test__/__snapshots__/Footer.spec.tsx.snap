// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Footer Test get default Footer no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  padding: 1rem;
}

.c3 {
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  font-size: 16px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #2E2E2E;
}

.c4 {
  font-size: 16px;
  line-height: 1.5;
  font-family: 'Open Sans';
  color: #2E2E2E;
}

.c5 {
  font-size: 16px;
  line-height: 1.5;
  font-family: 'Open Sans';
  background-image: linear-gradient( transparent calc(100% - 1px), black 1px );
  background-repeat: no-repeat;
  background-size: 0% 100%;
  -webkit-transition: background-size 0.5s;
  transition: background-size 0.5s;
  color: #2E2E2E;
}

.c5:hover {
  background-size: 100% 100%;
  cursor: pointer;
}

.c6 {
  color: #2E2E2E;
}

<body>
    <div>
      <div
        class="c0"
        display="flex"
        width="1"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
          >
            <div
              class="c1"
            >
              <p
                class="MuiTypography-root c2 MuiTypography-body1"
                customvariant="header3"
              >
                Questions
              </p>
              <div
                class="c3"
                display="flex"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/help"
                >
                  <span
                    class="MuiTypography-root c4 MuiTypography-overline"
                    customvariant="body1"
                  >
                    Help
                  </span>
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/trackorder"
                >
                  <span
                    class="MuiTypography-root c4 MuiTypography-overline"
                    customvariant="body1"
                  >
                    Track Order
                  </span>
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/returns"
                >
                  <span
                    class="MuiTypography-root c4 MuiTypography-overline"
                    customvariant="body1"
                  >
                    Return
                  </span>
                </a>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
          >
            <div
              class="c1"
            >
              <p
                class="MuiTypography-root c2 MuiTypography-body1"
                customvariant="header3"
              >
                What's in store
              </p>
              <div
                class="c3"
                display="flex"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/productlist?category=women"
                >
                  <span
                    class="MuiTypography-root c4 MuiTypography-overline"
                    customvariant="body1"
                  >
                    Women
                  </span>
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/productlist?category=men"
                >
                  <span
                    class="MuiTypography-root c4 MuiTypography-overline"
                    customvariant="body1"
                  >
                    Men
                  </span>
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/productlist"
                >
                  <span
                    class="MuiTypography-root c4 MuiTypography-overline"
                    customvariant="body1"
                  >
                    Product A to Z
                  </span>
                </a>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
          >
            <div
              class="c1"
            >
              <p
                class="MuiTypography-root c2 MuiTypography-body1"
                customvariant="header3"
              >
                Contact us
              </p>
              <div
                class="c3"
                display="flex"
              >
                <div
                  class=""
                >
                  <span
                    animation="1"
                    class="MuiTypography-root c5 MuiTypography-overline"
                    customvariant="body1"
                  >
                    Facebook
                  </span>
                </div>
                <div
                  class=""
                >
                  <span
                    animation="1"
                    class="MuiTypography-root c5 MuiTypography-overline"
                    customvariant="body1"
                  >
                    Twitter
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
          >
            <div
              class="c1"
            >
              <p
                class="MuiTypography-root c6 MuiTypography-body1"
              >
                Â©2020 shopify
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm usluC"
      display="flex"
      width="1"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
        >
          <div
            class="sc-AxjAm fJPldx"
          >
            <p
              class="MuiTypography-root sc-AxirZ bNdxLL MuiTypography-body1"
              customvariant="header3"
            >
              Questions
            </p>
            <div
              class="sc-AxjAm bYhklM"
              display="flex"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/help"
              >
                <span
                  class="MuiTypography-root sc-AxirZ iNyqdo MuiTypography-overline"
                  customvariant="body1"
                >
                  Help
                </span>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/trackorder"
              >
                <span
                  class="MuiTypography-root sc-AxirZ iNyqdo MuiTypography-overline"
                  customvariant="body1"
                >
                  Track Order
                </span>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/returns"
              >
                <span
                  class="MuiTypography-root sc-AxirZ iNyqdo MuiTypography-overline"
                  customvariant="body1"
                >
                  Return
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
        >
          <div
            class="sc-AxjAm fJPldx"
          >
            <p
              class="MuiTypography-root sc-AxirZ bNdxLL MuiTypography-body1"
              customvariant="header3"
            >
              What's in store
            </p>
            <div
              class="sc-AxjAm bYhklM"
              display="flex"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/productlist?category=women"
              >
                <span
                  class="MuiTypography-root sc-AxirZ iNyqdo MuiTypography-overline"
                  customvariant="body1"
                >
                  Women
                </span>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/productlist?category=men"
              >
                <span
                  class="MuiTypography-root sc-AxirZ iNyqdo MuiTypography-overline"
                  customvariant="body1"
                >
                  Men
                </span>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/productlist"
              >
                <span
                  class="MuiTypography-root sc-AxirZ iNyqdo MuiTypography-overline"
                  customvariant="body1"
                >
                  Product A to Z
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
        >
          <div
            class="sc-AxjAm fJPldx"
          >
            <p
              class="MuiTypography-root sc-AxirZ bNdxLL MuiTypography-body1"
              customvariant="header3"
            >
              Contact us
            </p>
            <div
              class="sc-AxjAm bYhklM"
              display="flex"
            >
              <div
                class="sc-AxjAm dgJqsq"
              >
                <span
                  animation="1"
                  class="MuiTypography-root sc-AxirZ ggGSAr MuiTypography-overline"
                  customvariant="body1"
                >
                  Facebook
                </span>
              </div>
              <div
                class="sc-AxjAm dgJqsq"
              >
                <span
                  animation="1"
                  class="MuiTypography-root sc-AxirZ ggGSAr MuiTypography-overline"
                  customvariant="body1"
                >
                  Twitter
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
        >
          <div
            class="sc-AxjAm fJPldx"
          >
            <p
              class="MuiTypography-root sc-AxirZ dJoWlE MuiTypography-body1"
            >
              Â©2020 shopify
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
